src/commands/mobile_app/upload.rs

```diff
@@ -77,3 +80,45 @@ fn validate_is_mobile_app(path: &Path) -> Result<()> {
         path.display()
     ))
 }
+
+/// Normalizes a mobile app file into a zip file to ensure consistent artifact parsing logic on the backend.
+fn normalize_mobile_app(path: &Path) -> Result<TempFile> {
+    let temp_file = TempFile::create()?;
+    let mut zip = ZipWriter::new(temp_file.open()?);
+
+    if path.is_file() {
```

Comment:
Should this handle the case where the user already zipped the file? Seems like this might re-zip it.